### PR TITLE
Fixes configuration for DotNet samples

### PR DESCRIPTION
**Pull request by rozele**
Previously, the sample configuration scripts (`sampleenv.ps1` and `sampleenv.sh`) would set the configuration values to environment variables.

The `Main.cs` shared file previously expected the configuration script to store a configuration file to the users home directory.

This changeset fixes the configuration scripts to write to a file, and changes the expected location to a file that is local to the VS solution.

## Description

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] If applicable, the code is properly documented.
- [x] The code builds without any errors.

### CODE REVIEW

**Comment by rozele** [comment]
> SB_SAMPLES_SUBSCRIPTIONNAME=Subscription1

Fix this.

---
Refers to: samples/DotNet/Microsoft.Azure.ServiceBus/common/azure-msg-config.properties:7 in a39c936. [](commit_id = a39c9368d2264c72747fe20be1f856e7497d02f0, deletion_comment = False)

**Comment by rozele** [comment]
> 

Add a comment.

---
Refers to: samples/DotNet/Microsoft.Azure.ServiceBus/common/.gitignore:2 in a39c936. [](commit_id = a39c9368d2264c72747fe20be1f856e7497d02f0, deletion_comment = False)

File: samples/DotNet/Microsoft.Azure.ServiceBus/scripts/sampleenv.sh [hunk 1]
```diff
@@ -36,10 +36,11 @@ then
      echo "$cx1" >&2
      exit 4
   fi
-  echo export SB_SAMPLES_CONNECTIONSTRING="$cx"
-  echo export SB_SAMPLES_QUEUENAME="BasicQueue"
-  echo export SB_SAMPLES_TOPICNAME="BasicTopic"
-  echo export SB_SAMPLES_SUBSCRIPTIONNAME="Subscription1"
+  configFile = "$( dirname "${BASH_SOURCE[0]}" )/../common/azure-msg-config.properties"
+  echo SB_SAMPLES_CONNECTIONSTRING="$cx"
+  echo SB_SAMPLES_QUEUENAME="BasicQueue"
```
Comment (rozele) [inline samples/DotNet/Microsoft.Azure.ServiceBus/scripts/sampleenv.sh]: Fix

File: samples/DotNet/Microsoft.ServiceBus.Messaging/.vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (rozele) [inline samples/DotNet/Microsoft.ServiceBus.Messaging/.vscode/settings.json]: Delete.

Comment (rozele) [reply samples/DotNet/Microsoft.ServiceBus.Messaging/.vscode/settings.json]: Add to .gitignore


---
In reply to: [180609636](https://github.com/Azure/azure-service-bus/pull/185#discussion_r180609636) [](ancestors = 180609636)

**Comment by rozele** [comment]
> SB_SAMPLES_QUEUENAME=BasicQueue

Fix.

---
Refers to: samples/DotNet/Microsoft.ServiceBus.Messaging/common/azure-msg-config.properties:3 in a39c936. [](commit_id = a39c9368d2264c72747fe20be1f856e7497d02f0, deletion_comment = False)

File: samples/DotNet/Microsoft.ServiceBus.Messaging/common/Main.targets [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<Project DefaultTargets="Compile"  
+    xmlns="http://schemas.microsoft.com/developer/msbuild/2003">  
+  <ItemGroup>
+    <Compile Include="$(MSBuildThisFileDirectory)\Main.cs" Link="Main.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="$(MSBuildThisFileDirectory)\azure-msg-config.properties" Link="azure-msg-config.properties">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (rozele) [inline samples/DotNet/Microsoft.ServiceBus.Messaging/common/Main.targets]: >CopyToOutputDirectory [](start = 7, length = 21)

Consolidate to one folder.

**Comment by rozele** [comment]
It's also nice to be able to edit `azure-msg-config.properties` directly from Visual Studio (since it's now a project resource).

**Comment by clemensv** [comment]
I'd like to understand the actual problem you're fixing. 

The point of having this in the user profile is that the same set of variables can be used from all projects even if you're making a copy in a branch repo elsewhere and you can still use the shared setup from the file in your profile. The environment variables are an override mechanism for what's in the file so you can modify your environment for a scenario without modifying the file.

**Comment by rozele** [comment]
@clemensv - the problem I'm fixing is that this file https://github.com/Azure/azure-service-bus/blob/master/samples/DotNet/Microsoft.Azure.ServiceBus/scripts/sampleenv.ps1 does not ever write the configuration to a file in the user directory.

I understand that you'd like to share a single file for all samples. Can I instead suggest that we make this change so that there is a single `common` and `scripts` folder under the root `samples` folder that is used for Java and DotNet samples?

**Comment by clemensv** [comment]
I don't want that file in the samples tree, at all. The file is a shared config option that you can use if you like and that should work irrespective of what your current branch is; therefore it's not inside the tree. The script is, in fact, meant to pick up the settings from your current deployment and stash them into the environment variables and you go from there; it's intentionally volatile. I'm okay with a modification to the script so that it actually optionally writes the file.

What's wrong with how Main.cs is referenced? Is that an artifact of the latest VS?

**Comment by rozele** [comment]
Okay - nevermind. I got it. You can ignore this change. There must have been some other problem I ran into while initializing the configuration. I now see where the environment variables are being created.

**Comment by rozele** [comment]
@clemensv - can you also look at #183? I had some issues with the AzureDeploy buttons and the READMEs.

**Comment by rozele** [comment]
@clemensv - this may be a dumb question, but where do I need to run the powershell script from to ensure Visual Studio 2017 will pick up the environment variables?

**Comment by clemensv** [comment]
Regarding the question of how to pick up th env variables: start devenv from the context where they have been set

**Pull request closed without merging**